monai/transforms/inverse.py

```diff
@@ -0,0 +1,113 @@
+    since it only stores strings, numbers and arrays.
+
+    We currently check that the ``id()`` of the transform is the same in the forward and
+    inverse directions. This is a useful check to ensure that the inverses are being
+    processed in the correct order. However, this may cause issues if the ``id()`` of the
+    object changes (such as multiprocessing on Windows). If you feel this issue affects
+    you, please raise a GitHub issue.
+
+    Note to developers: When converting a transform to an invertible transform, you need to:
+
+        #. Inherit from this class.
+        #. In ``__call__``, add a call to ``push_transform``.
+        #. Any extra information that might be needed for the inverse can be included with the
+           dictionary ``extra_info``. This dictionary should have the same keys regardless of
+           whether ``do_transform`` was `True` or `False` and can only contain objects that are
+           accepted in pytorch data loader's collate function (e.g., `None` is not allowed).
+        #. Implement an ``inverse`` method. Make sure that after performing the inverse,
+           ``pop_transform`` is called.
+
+    """
+
+    def push_transform(
+        self,
+        data: dict,
+        key: Hashable,
+        extra_info: Optional[dict] = None,
+        orig_size: Optional[Tuple] = None,
+    ) -> None:
+        """Append to list of applied transforms for that key."""
+        key_transform = str(key) + InverseKeys.KEY_SUFFIX.value
+        info = {
+            InverseKeys.CLASS_NAME.value: self.__class__.__name__,
+            InverseKeys.ID.value: id(self),
+            InverseKeys.ORIG_SIZE.value: orig_size or data[key].shape[1:],
+        }
+        if extra_info is not None:
+            info[InverseKeys.EXTRA_INFO.value] = extra_info
+        # If class is randomizable transform, store whether the transform was actually performed (based on `prob`)
+        if isinstance(self, RandomizableTransform):
+            info[InverseKeys.DO_TRANSFORM.value] = self._do_transform
+        # If this is the first, create list
+        if key_transform not in data:
+            data[key_transform] = []
+        data[key_transform].append(info)
+
+    def check_transforms_match(self, transform: dict) -> None:
+        """Check transforms are of same instance."""
+        if transform[InverseKeys.ID.value] != id(self):
+            raise RuntimeError("Should inverse most recently applied invertible transform first")
+
+    def get_most_recent_transform(self, data: dict, key: Hashable) -> dict:
+        """Get most recent transform."""
+        transform = dict(data[str(key) + InverseKeys.KEY_SUFFIX.value][-1])
+        self.check_transforms_match(transform)
+        return transform
+
+    def pop_transform(self, data: dict, key: Hashable) -> None:
+        """Remove most recent transform."""
+        data[str(key) + InverseKeys.KEY_SUFFIX.value].pop()
```

Comment:
had another look at this class, perhaps `get_most_recent_transform` could be merged into `pop_transform`?

```suggestion
    def pop_transform(self, data: dict, key: Hashable) -> None:
        """Removes and returns most recent transform."""
        xform = dict(data[str(key) + InverseKeys.KEY_SUFFIX.value].pop())
        self.check_transforms_match(xform)
        return xform
```

Reply:
I don't really want to pop when getting the most recent transform as I only want to remove it from the list once the inverse has been successfully applied.

Reply:
ok, at the moment I find the methods are always used together... it seems it's safe to merge them, or do you mean do you want to recover from some inversion failure in the future?